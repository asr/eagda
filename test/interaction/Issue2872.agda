open import Agda.Builtin.List
open import Agda.Builtin.Char

foo : Char → List Char
foo c = c ∷ []

case-split-test : List Char → List Char
case-split-test ('-' ∷ ls) with (foo '+')
... | foo-ls = {!foo-ls!}
case-split-test ls = ls

-- WAS: case splitting on foo-ls produces:
--   case-split-test (.'-' ∷ ls) | [] = ?
--   case-split-test (.'-' ∷ ls) | x ∷ foo-ls = ?
-- SHOULD: not put dots in front of literal patterns:
--   case-split-test ('-' ∷ ls) | [] = {!!}
--   case-split-test ('-' ∷ ls) | x ∷ foo-ls = {!!}
