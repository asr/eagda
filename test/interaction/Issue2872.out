(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : List Char " nil)
((last . 1) . (agda2-goals-action '(0)))
((last . 2) . (agda2-make-case-action '("case-split-test ('-' ∷ ls) | [] = ?" "case-split-test ('-' ∷ ls) | x ∷ foo-ls = ?")))
((last . 1) . (agda2-goals-action '(0)))
